### DISCUSSION TITLE
How to use different compression for different pages in multipage tiff

### DISCUSSION THREAD

**Discussion by Ovini** [dlemstra/Magick.NET#697] [Help] [answered]
I want to create multi page tiff file using a jpeg and a tiff file. The output should be a tiff file with page 1 in jpeg and page 2 in tiff. When I try it with below code page 2 also becomes a jpeg not a tiff. `

         using (MagickImageCollection images = new MagickImageCollection()){

                MagickImage firstFrame = new MagickImage(@"c:\dips\img1.jpg");
                firstFrame.Format = MagickFormat.Tiff;
                firstFrame.Settings.Compression = CompressionMethod.Group4;
                images.Add(firstFrame);

                MagickImage secondFrame = new MagickImage(@"c:\dips\img2.tif");
                secondFrame.Format = MagickFormat.Tiff;
                secondFrame.Settings.Compression = CompressionMethod.Group4;
                images.Add(secondFrame);

                images.Write(@"C:\dips\out.tiff", MagickFormat.Tiff);
         }
 `
I used https://www.irfanview.com/ to view the page properties of tiff file. Below is the properties of tiff pages. both pages shows the compression as jpeg where I want the 2nd page to be 'CCITT Group 4 Fax Encoding' which is tiff.
![image](https://user-images.githubusercontent.com/9262155/87770230-5e113380-c83c-11ea-9718-cd3eb8520d75.png)
Please Help.

**Comment by dlemstra** [comment]
Thanks for reporting this. It seems that you found an issue in the Magick.NET library. The settings of the first image are used when the tiff file is writen. But it should be possible to set the compression based on the compression of the image and not it's settings. This is not possible with the current API and I am not planning to make `image.Compression` writable again. I suspect that I will need to introduce a new method to change the compression of the image with a clear intelisense comment that you should only use that method if you know what you are doing.

**Comment by dlemstra** [comment, marked as answer]
In the next version of Magick.NET you will be able to write the file with the following code:

```C#
using (MagickImageCollection images = new MagickImageCollection())
{
    MagickImage firstFrame = new MagickImage(@"c:\dips\img1.jpg");
    firstFrame.Format = MagickFormat.Tiff;
    firstFrame.SetCompression(CompressionMethod.JPEG);
    images.Add(firstFrame);

    MagickImage secondFrame = new MagickImage(@"c:\dips\img2.tif");
    secondFrame.Format = MagickFormat.Tiff;
    secondFrame.SetCompression(CompressionMethod.Group4);
    images.Add(secondFrame);

    images.Write(@"C:\dips\out.tiff", new TiffWriteDefines
    {
        PreserveCompression = true,
    });
}
```

**Reply by Ovini** [reply]
Thank you very much. Any idea when will be the next release?

**Reply by dlemstra** [reply]
The new release has just been published.

**Reply by Ovini** [reply]
Thanks it works now. 🥇

**Comment by Ovini** [comment]
One small thing... I want the compression of above jpeg to be 'Old-style JPEG (in TIFF)'. Isn't it supported by the library?
![image](https://user-images.githubusercontent.com/9262155/88012510-bf385000-cb37-11ea-9b23-f7697baf4dbb.png)

**Reply by dlemstra** [reply]
We are using the libtiff library and that only supports decoding the files with OJPEG compression. An error will be raised by the library when this is tried: https://gitlab.com/libtiff/libtiff/-/blob/4c750d023832acb6c43d540f36e91d424018f15c/libtiff/tif_ojpeg.c#L913.
